### ISSUE TITLE
currentUserCredentials doesn't return expected authenticated value

### ISSUE THREAD

**Issue by RBrNx** [aws-amplify/amplify-js#7398]
**Describe the bug**
Calling `Auth.currentUserCredentials` after signing into Cognito with `Auth.signIn()` returns valid credentials, but for an unauthenticated user; `autheticated` is set to `false`.

**To Reproduce**

1. Sign into Cognito with a valid Username/Password - `await Auth.signIn(username, password)`
2. Request Credentials with `await Auth.currentUserCredentials()`
3. Returned Credentials with have the `authenticated` property set to `false`

**Expected behavior**
After signing into Cognito with a valid username/password, a call to `Auth.currentUserCredentials()` should return an authenticated set of Credentials.

**Code Snippet**
```javascript
  const attemptLogin = async () => {
    const beforeAuth = await Auth.currentUserCredentials();
    console.log({ beforeAuth });

    const user = await Auth.signIn(username, password);
    console.log({ user });

    const afterAuth = await Auth.currentUserCredentials();
    console.log({ afterAuth });
  };

//Object {
//  "beforeAuth": Object {
//    ...
//    "authenticated": false,
//    "expiration": 2020-12-12T18:58:06.000Z,
//
//  },
//}

//Object {
//  "user": CognitoUser {
//  ...
//  "signInUserSession": CognitoUserSession {
//      "accessToken": CognitoAccessToken {
//        "jwtToken": "<token>",
//        "payload": Object {
//           ...
//        },
//      },
//      "idToken": CognitoIdToken {
//        "jwtToken": "<token>",
//        "payload": Object {
//          ...
//        },
//      },
//      "refreshToken": CognitoRefreshToken {
//        "token": "<token>",
//      },
//    },
//    "username": "5fd3b1283126ba63f1f26964",
//  },

//Object {
//  "afterAuth": Object {
//    ...
//    "authenticated": false,
//    "expiration": 2020-12-12T18:58:07.000Z,
//}
```
 

<details>
  <summary><strong>Environment</strong></summary>

```
System:
    OS: macOS 11.0.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 913.86 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.18.1 - ~/.nvm/versions/node/v12.18.1/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.7 - ~/.nvm/versions/node/v12.18.1/bin/npm
  Browsers:
    Brave Browser: 87.1.17.75
    Chrome: 87.0.4280.88
    Firefox Developer Edition: 78.0
    Safari: 14.0.1
  npmPackages:
    @apollo/react-hooks: ^3.1.3 => 3.1.5 
    @babel/core: ^7.0.0 => 7.11.1 
    @expo/vector-icons: ^10.0.0 => 10.2.0 
    @react-native-community/masked-view: 0.1.10 => 0.1.10 
    @react-native-community/netinfo: 5.9.6 => 5.9.6 
    @react-navigation/bottom-tabs: ^5.1.1 => 5.8.0 
    @react-navigation/native: ^5.0.8 => 5.7.3 
    @react-navigation/stack: ^5.1.0 => 5.9.0 
    agnostic-aws-signature: ^1.0.4 => 1.0.4 
    apollo-boost: ^0.4.7 => 0.4.9 
    apollo-cache-inmemory: ^1.6.6 => 1.6.6 
    apollo-client: ^2.6.10 => 2.6.10 
    apollo-link: ^1.2.14 => 1.2.14 
    apollo-link-error: ^1.1.13 => 1.1.13 
    apollo-link-http: ^1.5.17 => 1.5.17 
    aws-amplify: ^3.3.12 => 3.3.12 
    babel-eslint: ^10.1.0 => 10.1.0 
    babel-plugin-macros: ^2.8.0 => 2.8.0 
    babel-preset-expo: ^8.3.0 => 8.3.0 
    color: ^3.1.2 => 3.1.2 
    eslint: ^6.8.0 => 6.8.0 
    eslint-config-airbnb: ^18.0.1 => 18.2.0 
    eslint-config-prettier: ^6.10.0 => 6.11.0 
    eslint-plugin-import: ^2.22.1 => 2.22.1 
    eslint-plugin-jsx-a11y: ^6.2.3 => 6.3.1 
    eslint-plugin-prettier: ^3.1.2 => 3.1.4 
    eslint-plugin-react: ^7.19.0 => 7.20.6 
    eslint-plugin-react-hooks: ^2.5.0 => 2.5.1 
    expo: ^39.0.0 => 39.0.3 
    expo-av: ~8.6.0 => 8.6.0 
    expo-constants: ~9.2.0 => 9.2.0 
    expo-splash-screen: ~0.6.1 => 0.6.1 
    expo-status-bar: ~1.0.2 => 1.0.2 
    graphql: ^14.6.0 => 14.7.0 
    graphql-tag: ^2.10.3 => 2.11.0 
    graphql.macro: ^1.4.2 => 1.4.2 
    jest: ^26.0.1 => 26.4.0 
    lottie-react-native: ~2.6.1 => 2.6.1 
    prettier: ^1.19.1 => 1.19.1 
    react: 16.13.1 => 16.13.1 
    react-dom: 16.13.1 => 16.13.1 
    react-native: https://github.com/expo/react-native/archive/sdk-39.0.0.tar.gz => 0.63.2 
    react-native-appearance: ~0.3.3 => 0.3.4 
    react-native-gesture-handler: ~1.7.0 => 1.7.0 
    react-native-modal: ^11.5.6 => 11.5.6 
    react-native-reanimated: ~1.13.0 => 1.13.1 
    react-native-safe-area-context: 3.1.4 => 3.1.4 
    react-native-screens: ~2.10.1 => 2.10.1 
    react-native-svg: 12.1.0 => 12.1.0 
    react-native-url-polyfill: ^1.1.2 => 1.2.0 
    react-native-web: ~0.13.7 => 0.13.13 
    reanimated-bottom-sheet: ^1.0.0-alpha.19 => 1.0.0-alpha.22 
  npmGlobalPackages:
    @nestjs/cli: 7.4.1
    @vue/cli: 4.4.6
    expo-cli: 4.0.3
    node-gyp: 7.1.2
    npm: 6.14.7
    puppeteer-lottie-cli: 1.0.9
    serverless: 2.9.0
```

</details>

**Comment by amhinson** [comment]
@RBrNx I can't seem to replicate the issue you are seeing. I'm seeing the expected behavior of `authenticated` being `true` after `Auth.signIn(...)` with brand new React and React Native apps and Amplify freshly setup and configured. Would you try creating a new app with Amplify and see if you're still seeing the same behavior? If so, can you share the exact steps you selected when setting up the app with Amplify CLI?

**Comment by RBrNx** [comment]
Hey @amhinson, thanks for taking a look at this issue.

So I can definitely say that I didn't use the Amplify CLI at all when setting up this project (mostly because I only need the `Auth` API).
Instead I simply have it configured with
```javascript
Amplify.configure({
  Auth: {
    region: awsConfig.cognito.REGION,
    userPoolId: awsConfig.cognito.USER_POOL_ID,
    identityPoolId: awsConfig.cognito.IDENTITY_POOL_ID,
    userPoolWebClientId: awsConfig.cognito.APP_CLIENT_ID,
  },
});
```

where `awsConfig` is the following object:
```javascript
export default {
  cognito: {
    REGION: 'eu-west-2',
    USER_POOL_ID: 'eu-west-2_Z34drbGTL',
    APP_CLIENT_ID: '1llgl3u1khd1t8v1t600vcr2ri',
    IDENTITY_POOL_ID: 'eu-west-2:59c7f5f5-851e-4186-b65f-70e78dc0f963',
  },
};
```

I'm happy to create a new project and try the same configuration though if you think it will help?

**Comment by amhinson** [comment]
Ah ok thanks for the info! It could be helpful to try creating a new project with the CLI and then (assuming the CLI project works as expected) compare your existing resources and particularly the IAM roles. Let me know what you find! 🙂

**Comment by elorzafe** [comment]
@RBrNx 

Do you have enabled MFA?

**Comment by RBrNx** [comment]
Hey @amhinson, so I tried creating a new project with the Amplify CLI, following these instructions; unfortunately I still seem to have the same issue.

Would it help if I pushed the new project to a Repository and shared it with you?

I have also tried creating a new Cognito Pool with all of the defaults set. I can log into this new User Pool successfully, but upon calling `await Auth.currentUserCredentials()` I get `"No Cognito Identity pool provided for unauthenticated access"`. This error somewhat makes sense as Cognito Pools by default don't allow unauthenticated access, but I should be authenticated as I successfully managed to log in?

@elorzafe I do not have MFA Enabled on either the original User Pool, nor the test one I just created.

**Comment by RBrNx** [comment]
So, I just realised I missed running `amplify init` when I was following the installation instructions (d'oh). With an amplify created stack I am now seeing `authenticated` return as `true` correctly!

It must be an issue with the way my other stack was set up, I'll start by checking those IAM roles you mentioned but is there  anywhere else you would suggest to look?

**Comment by amhinson** [comment]
Ah ok thanks for looking into in @RBrNx! Those roles would be the main thing to check, but it also couldn't hurt to just simply go through all of the Cognito settings to compare between the two if you are still having issues after checking the roles.

**Comment by elorzafe** [comment]
@RBrNx when you created your identity pool, how did you enabled your userpool as Authentication provider?

**Comment by RBrNx** [comment]
@elorzafe it's created for me as part of a Serverless Framework stack, but I believe it puts my User Pool ID and an App Client ID into the 'Cognito' Tab under 'Authentication Providers' in the Identity Pool settings.

**Comment by RBrNx** [comment]
So I spent some time comparing the IAM Roles created by Amplify, and the ones created by my Serverless config; there were a few differences but nothing that seemed to affect my issue.

I then went and started comparing the User Pools and then I noticed that the Amplify-created Identity Pool had two entries under "Authentication Providers", one for each client. Turns out I had only set up my Identity Pool with a single Authentication Provider, which happened to be my API Client instead of my Mobile Client! After adding in an entry for my Mobile Client (which is the one I was using to sign into Cognito) it now works as expected!

Thanks for the help @amhinson @elorzafe!
